Fix overflow edge case in sequence of balls volume method

dymil opened the pull request:
Force rnum>=1, avoiding integer overflow when rnum-1 is passed into RandomPointGenerator::apply a few lines later, which hoovers up RAM and time. This edge case can be observed with a degenerate polytope (I tried a 1D interval) or with extreme choices of `error` and `n_threads`.

dymil committed:
Fix rnum=0 causing integer overflow in SoB volume

rnum-1 is passed into RandomPointGenerator::apply a few lines later

vissarion reviewed (commented):
Thanks @dymil for this PR! I am OK with merging of this change but please add the edge cases you mention in the test suite (this will eventually increase the coverage, notice that the CI is complaining about coverage decrease).

vissarion commented:
You can add tests here https://github.com/GeomScale/volesti/blob/develop/test/volume_sob_hpolytope.cpp for (sequence of balls volume algorithm)

dymil committed:
Tweak bugfix and add 1D volume test
Force rnum>=2 rather than >=1,
so that all three walk methods succeed for 1D case.
Small rnum's due to larger error or n_threads may fail in other ways
in higher dimensions, e.g., empty balls check

dymil commented:
I added a 1D test case which fails (times out) with the old code but passes with the updated (after tweaking the fix to ensure `rnum>=2` instead of 1). I also tried a 2D test case with large `error` or `n_threads` and those indeed trigger the overflow but, even with the fix, `balls` would be empty or something else would go wrong with a small `rnum`. The fix presumably needs to scale with dimension but I'm not sure exactly how, and, in any case, I find failing an assertion superior to simply hanging.

Alternative fixes which might be more suitable include shrinking n_threads if needed (`rnum` may still need to be increased if it was already small) or to make an earlier assertion so that the caller is better informed of the ultimate cause of the issue, rather than silently plugging the issue.
